server/nova-components/CreateFavorite/resources/js/components/form/form.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+
+    <div>
+            <div class="form-container">
+                <div class="error-box" v-if="error!=''">Error : {{error}}</div>
+                <p class = "form-title">{{title}}</p>
+                <form v-on:submit.prevent="validateForm" >
+                    <form-input type="text" name="url"         :req="true"   title="URL" :value="info.url"    :handleChange="handleChange" /> 
+                    <select-input name="user" :options="users" :req="true"  :value="info.user"  :handleChange="handleChange" />
+                    <form-input type="text" name="title"           title="Title" :value="info.title"     :handleChange="handleChange"  /> 
+                    
+                    <form-input type="text" name="author"      title="Author" :value="info.author"     :handleChange="handleChange" /> 
+                    <form-input type="text" name="description" title="Description" :value="info.description"    :handleChange="handleChange" /> 
+                    <form-input type="text" name="urlToImage"  title="Image" :value="info.urlToImage"    :handleChange="handleChange" /> 
+
+                    <form-input type="date" name="publishedAt" title="Publish Date" :value="info.publishedAt"  :handleChange="handleChange" /> 
+                    <button-input :disabled="canSubmit" type="submit"  text="Create" />
+                </form>
+            </div>
+        </div>
+        
+</template>
+
+
+<script>
+import {buttonInput,formInput,selectInput} from './inputs/index.js'
+import { parseNovaApi } from './../../helpers/index.js'
+import './form.css'
+
+    export default {
+        components: { formInput ,buttonInput,selectInput},
+        name: "form",
+        props:{
+            title:{
+                type:String,
+                default:"Form"
+            }
+        },
+        data(){
+            return{
+                info:{
+                title:"",
+                author:"",
+                description:"",
+                urlToImage:"",
+                publishedAt:"",
+                url:"",
+                user:0
+            },
+                users:[],
+                error:''
+            }
+        },
+        computed: {
+            canSubmit: function(){
+                return !(this.info.url.length && this.info.user  )
```

Comment:
1- reverse condition order
2- follow a consistent convention when you declare functions